src/main/java/org/jenkinsci/plugins/github_branch_source/Connector.java

```diff
@@ -102,11 +102,6 @@
 
         gb.withRateLimitHandler(CUSTOMIZED);
         OkHttpClient client = new OkHttpClient().setProxy(getProxy(defaultIfBlank(apiUrl, GITHUB_URL)));
-        client.setCache(GitHubClientCacheOps.toCacheDir().apply(config));
-        if (config.getClientCacheSize() > 0) {
-            Cache cache = toCacheDir().apply(config);
-            client.setCache(cache);
-        }
```

Comment:
@jglick Flushing the cache on incoming webhooks has some problems:

1. Webhooks entry points are `GitHubWebhookListenerImpl` and `PullRequestGHEventSubscriber`, once inside `retrieve` there is no way to know if the indexing was triggered by a webhook or not. But flushing the cache in the subscribers does not fix the issue: two consecutive webhooks could trigger two calls to `github.getRepositories()` without any cache invalidation in between (because both cache invalidation calls are previous to the API call).
2. If two branch indexing are manually triggered more or less near in time and a commit happens in the middle, the cache issue will hit again.
3. The invalidation has to be done by reflection as `urlFactory` is private in `OkHttpConnector` (not really a problem, but really ugly, I would avoid it if possible).

Said that, the cache invalidation has to be done just before the API call (`github.getRepositories()` inside `GitHubSCMSource#retrieve`) which is near to be complete cache disabling.

Moreover, the API call to get all branches in all repositories in probably the only one we would interested on caching but it the one we can not cache! All other API calls in the plugin are really lightweight or not frequently done, so what's the real benefit of invalidating the cache selectively?